src/main/java/org/opensearch/ad/ml/EntityModel.java

```diff
@@ -61,6 +62,9 @@ public EntityModel(Entity entity, Queue<double[]> samples, RandomCutForest rcf,
     }
 
     public void addSample(double[] sample) {
+        if (this.samples == null) {
```

Comment:
Should we add this null checking in `clear(0` function as well?

Reply:
yes, fixed.